### PR TITLE
Refine implementation and usage of `throwAsUncheckedException(..)`

### PR DESCRIPTION
**Pull request by bjmi**
Use `throw` with `throwAsUncheckedException(..)`.

Reason:

`ExceptionUtils.throwAsUncheckedException(t)` emits the following warning in IntelliJ:

> Result of 'throwAsUncheckedException()' not thrown

and `throw ExceptionUtils.throwAsUncheckedException(t)` also emits a warning:

> Dereference of 'ExceptionUtils.throwAsUncheckedException(t)' may produce 'NullPointerException' 

Therefore following recommendation: `throwAsUncheckedException` shouldn't return a null value.

---

I hereby agree to the terms of the [JUnit Contributor License Agreement](https://github.com/junit-team/junit5/blob/002a0052926ddee57cf90580fa49bc37e5a72427/CONTRIBUTING.md#junit-contributor-license-agreement).

### CODE REVIEW

File: junit-jupiter-engine/src/test/java/org/junit/jupiter/api/AssertTimeoutAssertionsTests.java [hunk 1]
```diff
@@ -158,11 +153,11 @@ void assertTimeoutWithMessageSupplierForSupplierThatCompletesAfterTheTimeout() {
 	 * Take a nap for 100 milliseconds.
 	 */
 	private void nap() throws InterruptedException {
-		long start = System.currentTimeMillis();
+		long start = System.nanoTime();
 		// workaround for imprecise clocks (yes, Windows, I'm talking about you)
 		do {
 			Thread.sleep(100);
-		} while (System.currentTimeMillis() - start < 100);
+		} while (System.nanoTime() - start < 100_000_000L);
```
Comment (sbrannen) [inline junit-jupiter-engine/src/test/java/org/junit/jupiter/api/AssertTimeoutAssertionsTests.java:160]: Since this is not related to `throwAsUncheckedException()`, please move these changes to a separate PR with the rationale for the change.

File: junit-jupiter-engine/src/test/kotlin/org/junit/jupiter/api/KotlinAssertTimeoutAssertionsTests.kt [hunk 2]
```diff
@@ -270,11 +270,11 @@ internal class KotlinAssertTimeoutAssertionsTests {
      * Take a nap for 100 milliseconds.
      */
     private fun nap() {
-        val start = System.currentTimeMillis()
+        val start = System.nanoTime()
         // workaround for imprecise clocks (yes, Windows, I'm talking about you)
         do {
             Thread.sleep(100)
-        } while (System.currentTimeMillis() - start < 100)
+        } while (System.nanoTime() - start < 100_000_000)
```
Comment (sbrannen) [inline junit-jupiter-engine/src/test/kotlin/org/junit/jupiter/api/KotlinAssertTimeoutAssertionsTests.kt:277]: Since this is not related to `throwAsUncheckedException()`, please move these changes to a separate PR with the rationale for the change.

**Review by sbrannen — CHANGES_REQUESTED** [review]
Thanks for the PR.

I've requested that unrelated changes be moved to a separate PR.

File: junit-jupiter-engine/src/test/java/org/junit/jupiter/api/AssertTimeoutAssertionsTests.java (hunk 1 above)
Comment (bjmi) [reply junit-jupiter-engine/src/test/java/org/junit/jupiter/api/AssertTimeoutAssertionsTests.java]: Reverted.

File: junit-jupiter-engine/src/test/kotlin/org/junit/jupiter/api/KotlinAssertTimeoutAssertionsTests.kt (hunk 2 above)
Comment (bjmi) [reply junit-jupiter-engine/src/test/kotlin/org/junit/jupiter/api/KotlinAssertTimeoutAssertionsTests.kt]: Reverted.

**Comment by sbrannen** [comment]
This has been merged into `main`.

Thanks
